source/core/WorldGrid.hpp

```diff
@@ -193,6 +193,13 @@ namespace cse491 {
       return true;
     }
 
+    /// @brief Allows the system to replace any cell with a new cell type
+    /// @param grid 
+    /// @param newType 
+    void SetCell (GridPosition grid, int newType) {
+      int pos = (int)(width * grid.GetY()) + grid.GetX();
```

Comment:
You should avoid C-style cases.  Also, why is `pos` an `int`?  It should be `size_t`, which wouldn't involve casting.  Likewise `newType` (which should be shifted to `new_type`, but we only talked about that after your wrote this code) should be a `size_t`, since that's the type stored by `WorldGrid`.

Reply:
Also, we shouldn't put the conversion of x and y to a position in more than one spot.  There is a helper function `ToIndex` that you can use for this (so we only need to update it in one spot if something changes), or you can use one of the existing functions like `At()`.  Since I don't think we'll keep this function, it doesn't matter as much here, but you should always link in to existing functionality when possible rather than duplicating effort.